src/zjs_web_sockets.c

```diff
@@ -0,0 +1,793 @@
+        return;
+    }
+
+    // create the accept response
+    char *send_data = zjs_malloc(strlen(accept_header) + strlen(con->accept_key) + strlen(proto) + 32);
+    memset(send_data, 0, strlen(accept_header) + strlen(con->accept_key) + strlen(proto) + 32);
+    memcpy(send_data, accept_header, strlen(accept_header));
+    strcat(send_data, con->accept_key);
+    strcat(send_data, "\r\n");
+    strcat(send_data, "Sec-WebSocket-Protocol: ");
+    strcat(send_data, proto);
+    strcat(send_data, "\r\n\r\n\0");
+
+    DBG_PRINT("Sending accept packet\n");
+    tcp_send(con->tcp_sock, send_data, strlen(send_data));
+    con->state = AWAITING_ACCEPT;
+    zjs_free(send_data);
+    zjs_free(con->accept_key);
+    con->accept_key = NULL;
+}
+
+static void tcp_accepted(struct net_context *context,
+                         struct sockaddr *addr,
+                         socklen_t addrlen,
+                         int error,
+                         void *user_data)
+{
+    int ret;
+    server_handle_t *handle = (server_handle_t *)user_data;
+    DBG_PRINT("connection made, context %p error %d\n", context, error);
+    ws_connection_t *new = zjs_malloc(sizeof(ws_connection_t));
+    memset(new, 0, sizeof(ws_connection_t));
+    new->tcp_sock = context;
+    new->rbuf = zjs_malloc(DEFAULT_WS_BUFFER_SIZE);
+    memset(new->rbuf, 0, sizeof(DEFAULT_WS_BUFFER_SIZE));
+    new->rptr = new->wptr = new->rbuf;
+    new->server = handle->server;
+    new->trigger_msg_id = zjs_add_c_callback(new, trigger_data);
+    if (jerry_value_is_function(handle->accept_handler)) {
+        new->accept_handler_id = zjs_add_callback(handle->accept_handler,
+                                                  new->server_handle->server,
+                                                  new,
+                                                  post_accept_handler);
+    } else {
+        new->accept_handler_id = -1;
+    }
+    new->next = connections;
+    connections = new;
+
+    ret = net_context_recv(context, tcp_received, 0, new);
+    if (ret < 0) {
+        ERR_PRINT("Cannot receive TCP packet (family %d)\n",
+            net_context_get_family(context));
+        // this seems to mean the remote exists but the connection was not made
+        zjs_trigger_event(handle->server, "error", NULL, 0, NULL, NULL);
+        return;
+    }
+}
+
+static jerry_value_t ws_server(const jerry_value_t function_obj,
+                               const jerry_value_t this,
+                               const jerry_value_t argv[],
+                               const jerry_length_t argc)
+{
+    ZJS_VALIDATE_ARGS_OPTCOUNT(optcount, Z_OBJECT, Z_OPTIONAL Z_FUNCTION);
+
+    int ret;
+    double port = 0;
+    bool backlog = 0;
+    double max_payload = 0;
+
+    server_handle_t *handle = zjs_malloc(sizeof(server_handle_t));
+    memset(handle, 0, sizeof(server_handle_t));
+
+    jerry_value_t server = jerry_create_object();
+
+    zjs_obj_get_double(argv[0], "port", &port);
```

Comment:
if they don't exist then the port will/backlog/tracking/payload will be unchanged i.e. zero, which is fine.